x-pack/legacy/plugins/reporting/common/get_absolute_url.ts

```diff
@@ -5,27 +5,27 @@
  */
 
 import url from 'url';
-import { oncePerServer } from '../server/lib/once_per_server';
-import { ServerFacade } from '../types';
-
-function getAbsoluteUrlFn(server: ServerFacade) {
-  const config = server.config();
+import { AbsoluteURLFactory } from '../types';
 
+export const getAbsoluteUrlFactory = ({
+  protocol,
+  hostname,
+  port,
+  defaultBasePath,
+}: AbsoluteURLFactory) => {
```

Comment:
Can we rename this to `AbsoluteURLFactoryOptions`? I think that's more common for naming the type for an object argument to a function.